src/schemas/types.ts

```diff
@@ -1,6 +1,20 @@
 import { IBaseTextBasedFieldAttributes } from "src/context-providers";
+import * as yup from "yup";
+import { BaseSchemaHelper } from "./base-helper";
+import { TEXT_AREA_SCHEMA, TEXT_BASED_SCHEMA } from "./text-based-fields";
 
 export type IBaseTextBasedFieldValues = Omit<
     IBaseTextBasedFieldAttributes,
     "internalId"
 >;
+
+export type TTextBasedSchema = yup.InferType<
+    ReturnType<typeof TEXT_BASED_SCHEMA>
+>;
+
+export type TTextAreaSchema = yup.InferType<
+    ReturnType<typeof TEXT_AREA_SCHEMA>
+>;
+
+export type TOverallTextBasedSchema = BaseSchemaHelper.TBaseSchema &
+    (TTextBasedSchema | TTextAreaSchema);
```

Comment:
I am not quite sure what are we trying to achieve here, `BaseSchemaHelper.TBaseSchema` is based on text based schema but we still added `TTextBasedSchema` and `TTextAreaSchema` here?